modules/mod-deep-learning/ModelCard.h

```diff
@@ -0,0 +1,212 @@
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+
+   ModelCard.h
+   Hugo Flores Garcia
+
+******************************************************************/
+/**
+
+\class ModelCard
+\brief model metadata for deep learning models
+
+*/
+/*******************************************************************/
+
+#pragma once
+
+#include <torch/script.h>
+#include <torch/torch.h>
+
+#include <rapidjson/document.h>
+#include <rapidjson/schema.h>
+#include "rapidjson/prettywriter.h" 
+#include <rapidjson/writer.h>
+#include <wx/string.h>
+#include <wx/log.h>
+
+#include "MemoryX.h"
+#include "AudacityException.h"
+
+using Doc = rapidjson::Document;
+
+// this exception should be caught internally, but we 
+// derive from MessageBoxException just in case it needs to 
+// get handled by Audacity
+class InvalidModelCardDocument final : public MessageBoxException
+{
+public:
+   InvalidModelCardDocument(const TranslatableString msg, std::string trace): 
+                           m_msg(msg), 
+                           m_trace(std::move(trace)),
+                            MessageBoxException {
+                               ExceptionType::Internal,
+                               XO("Invalid Model Card Document")
+                            }
+   { 
+   }
+
+   // detailed internal error message
+   virtual const char* what() const throw () 
+   {
+      wxLogError(m_msg.Translation());
+      if (!m_trace.empty()) 
+         wxLogError(wxString(m_trace)); 
+      // TODO: also print the document
+      // TODO: check if document is nullptr
+      return m_msg.Translation().c_str();
+   }
+
+   // user facing message
+   virtual TranslatableString ErrorMessage() const
+      { return XO("Model Card Error: %s")
+               .Format(m_msg);}
+
+   const TranslatableString m_msg;
+   const std::string m_trace;
+};
+
+namespace parsers 
+{
+   Doc ParseString(const std::string &json);
+   Doc ParseFile(const std::string &path);
+}
+
+// validators that accept a default value are no-throw
+namespace validators
+{
+
+   std::string tryGetString(const std::string &key, const Doc& doc);
+   std::string tryGetString(const std::string &key, const Doc& doc, const std::string &defaultValue);
+
+   int tryGetInt(const std::string &key, const Doc& doc);
+   int tryGetInt(const std::string &key, const Doc& doc, int defaultValue);
+
+   std::vector<std::string> tryGetStringArray(const std::string &key, const Doc& doc, bool lower=false);
+                                             
+   uint64_t tryGetUint64(const std::string &key, const Doc& doc);
+   uint64_t tryGetUint64(const std::string &key, const Doc& doc, uint64_t defaultValue);
+
+}
+
+class DeepModelManager;
+
+class ModelCard final
+{
+public: 
+   ModelCard() = default;
+   ModelCard(const ModelCard&) = default;
+
+   // returns {author}/{name}
+   std::string GetRepoID() const;
+
+   bool IsSame(const ModelCard& other) const { return (*this) == other; };
```

Comment:
unnecessary `;`

Reply:
Should comment that only GetRepoID is meant to be significant in comparison, not the other fields:  I was surprised to learn this when I got to reviewing the implementation.

Now I wonder why you need operator ==.  It raises the question, could this type also usefully be hashable or comparable with `<` ?

Reply:
d19180bd489b299b7e7a8caee2c5a9f13bd2e689